### PR TITLE
Fixes very broken plant dna manipulator (gene_modder.dm)

### LINKED ISSUE #562: Plant DNA Manipulator clears all other plant data disks when used

### PR DESCRIPTION
**Pull request by yorii**
## About The Pull Request

Fixes the super broken machine into a fully functioning state.

## Why It's Good For The Game

Fixes #562 
Also fixes another issue with the disks not being renamed properly after the gene stored inside.
Also tweaks some more minor oversights and bugs.

## Changelog

:cl:
fix: plant data disks no longer lose their data to garbage collection
fix: plant data disks now receive proper names when saved
fix: plant dna manipulator now respects maximum gene values according to component levels
fix: plant dna manipulator can be wrenched and moved
/:cl:

### CODE REVIEW

File: modular_zubbers/code/modules/hydroponics/gene_modder.dm [hunk 1]
```diff
@@ -255,31 +256,34 @@
 	popup.open()
 
 
-/obj/machinery/plantgenes/Topic(href, list/href_list)
+/obj/machinery/plantgenes/Topic(var/href, var/list/href_list)
```
Comment (nevimer) [inline modular_zubbers/code/modules/hydroponics/gene_modder.dm:259]: will fail linters, no need to declare var in proc args

**Review by nevimer — CHANGES_REQUESTED** [review]

File: modular_zubbers/code/modules/hydroponics/gene_modder.dm (hunk 1 above)
Comment (yorii) [reply modular_zubbers/code/modules/hydroponics/gene_modder.dm:259]: Oh, that's why linters is failing? Weird, I'll change it right away!

**Comment by yorii** [comment]
readded the unique plant disk name proc as well

**Comment by yorii** [comment]
actually when testing now the name changing still doesn't work...

**Comment by yorii** [comment]
Nevermind it was just VSCode being dumb, it all works perfectly now!

**Comment by yorii** [comment]
Found a massive oversight in how the genes are read, clamped, and stored to disk, now fixed, there are no longer garbage collection triggers on orphaned genes and the minimum/maximum stored gene values according to tech tier is now also working. (it definitely wasn't before)

**Comment by yorii** [comment]
Oops forgot a test string :)

PR is finished for merging now, and thoroughly tested!

**Comment by yorii** [comment]
why tf are the linters failing again

**Comment by yorii** [comment]
@nevimer the code is working perfectly like this but I hate how it looks... any suggestion?

**Comment by nevimer** [comment]
I think it's fine

**Pull request merged**